contracts/interchain-token-service/src/contract.rs

```diff
@@ -186,6 +188,60 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             .into()
     }
 
+    fn interchain_token_id(env: &Env, sender: Address, salt: BytesN<32>) -> BytesN<32> {
+        env.crypto()
+            .keccak256(&(PREFIX_INTERCHAIN_TOKEN_ID, sender, salt).to_xdr(env))
+            .into()
+    }
+
+    fn deploy_interchain_token(
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        token_meta_data: TokenMetadata,
+        initial_supply: i128,
+        _minter: Address,
+    ) -> Result<BytesN<32>, ContractError> {
+        caller.require_auth();
+
+        let minter = if initial_supply > 0 {
+            env.current_contract_address()
+        } else {
```

Comment:
doesn't match EVM logic. We check if minter was provided, and only then make the check and set a minter
https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/InterchainTokenFactory.sol#L148